### PR TITLE
Input/Device: Refactor Gyro events handling

### LINKED ISSUE #9136: "Lock auto rotation to current orientation" is unchecked, grayed out
* KOReader version: 2022.05.1
* Device: Oasis

#### Issue

On the Gear-menu, under Screen, the option "Lock auto rotation to current orientation" is unchecked.  I cannot check it because it is grayed out.

Whenever I tilt my device a bit the rotation jumps all over the place!  This is incredibly frustrating.  I do not want my device to do things simply because I tilted it one way or another.

#### Steps to reproduce

Go to Gear->Screen, observe grayedness.

Rotate device, see the UI jump around.

### PR DESCRIPTION
**Pull request by NiLuJe**
Handle a single, custom protocol in Input, and move the platform-specific translation to Device implementations.

This allows us to get rid of a whole bunch of nastiness from Input, as the Kindle handling in particular was... not pretty ;).

Also, tweak how input events hooks can be built to allow for a single composite hook on Kobo, instead of repeating and nesting calls.

Only tested on Kobo so far (I don't have any other gyro-enabled device anyway ;p).

Depends on https://github.com/koreader/koreader-base/pull/1568 (re: #9873)

---

* Get rid of the `canToggleGSensor` Device cap, it's now mandatory for `hasGSensor` devices. (This means Kindles can now toggle the gyro, fix #9136).
* This means that `Device:toggleGSensor` is now implemented by `Generic`.
* Update the Screen & Gyro rotation constants to be clearer (c.f., https://github.com/koreader/koreader-base/pull/1568) (/!\ This might conceivably break some `rotation_map` user-patches).
* Input: Move the platform-specific gyro handling to Device implementations, and let Input only handle a single, custom protocol (`EV_MSC:MSC_GYRO`).
* Input: Refine the `rotation_map` disable method implemented in 43b021d37cc01a924740aba34ae337d813dbf540. Instead of directly poking at the internal field, use a new method, `disableRotationMap` (/!\ Again, this might break some `rotation_map` user-patches).
* Input: Minor tweaks to event adjust hooks to make them more modular, allowing the Kobo implementation to build and use a single composite hook. API compatibility maintained with wrappers.

- - -
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/9935)

### CODE REVIEW

File: frontend/device/input.lua [hunk 1]
```diff
@@ -1393,23 +1322,15 @@ function Input:inhibitInput(toggle)
...
-            if not self.device:isPocketBook() and not self.device:isAndroid() then
-                -- NOTE: PocketBook is a special snowflake, synthetic Power events are sent as EV_MSC.
-                --       Thankfully, that's all that EV_MSC is used for on that platform.
-                -- NOTE: Android, on the other hand, handles a *lot* of critical stuff over EV_MSC,
-                --       as it's used to communicate between Android and Lua land ;).
-                self._msc_ev_handler = self.handleMiscEv
-                self.handleMiscEv = self.voidEv
-            end
+        -- NOTE: We leave handleMiscEv alone, as EV_MSC is used for critical low-level stuff on some platforms (PB, Android),
+        --       and the only thing we might want to skip in there are gyro events anyway, which we'll handle separately.
```
Comment (rjd22) [inline frontend/device/input.lua:1330]: I would keep the specifics in as they are useful for grasping the scale of things.

Comment (NiLuJe) [reply frontend/device/input.lua:1330]: The previously more detailed comment, you mean?

Comment (rjd22) [reply frontend/device/input.lua:1330]: Yeah, knowing how much and for what something is used is often very helpful for deciding if you're going to touch it ;).

**Review by Frenzie — APPROVED** [review]
(approval relates to the principle, no opinion as to whether the Kindle stuff was correctly transplanted :-P)

**Review by NiLuJe — COMMENTED** [review]
> Reviewed 7 of 12 files at r1, 16 of 16 files at r2, 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/koreader/koreader/9935)* status: all files reviewed, 1 unresolved discussion (waiting on @pazos, @poire-z, and @rjd22)

**Review by rjd22 — APPROVED** [review]
Looks good to me. After this is merged I will make and new PR for the PocketBook gyro detection. For enabling and disabling the gyro I will let users do that in the config menu of PocketBook itself. That will require less supporting code in our side.

**Comment by NiLuJe** [comment]
> For enabling and disabling the gyro I will let users do that in the config menu of PocketBook itself.

If all goes well, the standard "ignore the events" approach implemented here should Just Work (TM) anyway ;).

**Pull request merged**